File: samples/client/3_1_0_unit_test/java/src/main/java/unit_test_api/schemas/validation/BooleanEnumValidator.java
package unit_test_api.schemas.validation;

import unit_test_api.configurations.SchemaConfiguration;
import unit_test_api.exceptions.ValidationException;

public interface BooleanEnumValidator <EnumType extends BooleanValueMethod> {
    boolean validate(EnumType arg, SchemaConfiguration configuration) throws ValidationException;
}

File: samples/client/petstore/kotlin/src/main/kotlin/org/openapijsonschematools/client/schemas/validation/BooleanEnumValidator.kt
package org.openapijsonschematools.client.schemas.validation;

import org.openapijsonschematools.client.configurations.SchemaConfiguration;
import org.openapijsonschematools.client.exceptions.ValidationException;

public interface BooleanEnumValidator <EnumType extends BooleanValueMethod> {
    boolean validate(EnumType arg, SchemaConfiguration configuration) throws ValidationException;
}